src/celeritas/global/ActionLauncher.device.hh

```diff
@@ -34,8 +34,9 @@ namespace
 /*!
  * Launch the given executor using thread ids in the thread_range
  */
-template<class F>
+template<class F, int threadsPerBlock, int blocksPerSm>
 __global__ void
+__launch_bounds__(threadsPerBlock, blocksPerSm)
```

Comment:
Remember that these have different meanings for CUDA and HIP: "minimum blocks per CU" vs "minimum warps per EU" (AMD has 4 EU per CU, and NVIDIA has 1). I think we should choose the latter as our argument since it's more general and convert in the CUDA case.

Reply:
Right it's not exactly the same semantics. I'm pretty sure CUDA has 4 EU (warp scheduler) per SM for almost all architectures except CC6.0 which has 2.

Reply:
I think the difference is that with NVIDIA, warps can move around execution units in an SM while that block is being executed (it's the block pinned to the SM), whereas with AMD warps are pinned to a specific EU. It's another layer of granularity.

Reply:
Is that so? I'll need to double check but I thought that a warp is bound to a scheduler in cuda as well.